rclcpp/include/rclcpp/executors/static_executor_entities_collector.hpp

```diff
@@ -64,9 +64,14 @@ class StaticExecutorEntitiesCollector final
   void
   init(
     rcl_wait_set_t * p_wait_set,
-    rclcpp::memory_strategy::MemoryStrategy::SharedPtr & memory_strategy,
+    rclcpp::memory_strategy::MemoryStrategy::SharedPtr memory_strategy,
```

Comment:
why changing this into `by value`? it does not seem to be related to the issue.

Reply:
Yes, it's not related to this issue, but this usage seems not good and not necessary.

http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r35-take-a-shared_ptrwidget-parameter-to-express-that-a-function-might-reseat-the-shared-pointer

Reply:
thanks i was just checking on that. i think it makes sense to use different object by here.